app/set/directives/collections/listing.directive.js

```diff
@@ -0,0 +1,208 @@
+    }
+
+    // Handle behaviour of the collection listing item click
+    // In Listing mode: a click will take the user to the collections page
+    // In Toggle mode: a click will select a given collection
+    function collectionClickHandler(collection) {
+        $scope.isToggleMode ? $scope.toggleCollection(collection.id) : $scope.goToCollection(collection);
+    }
+
+    // This feature is only available in Lisiting mode
+    // When a user clicks on a collection lisiting item
+    // they will be directed to the collection page
+    function goToCollection(collection) {
+        $scope.collectionListingVisible = false;
+        $location.path($scope.path + collection.id + '/' + collection.view);
+    }
+
+    // Toggle a post as selected or not
+    // This behaviour is specific to the Toggle mode of collection listing
+    // It allows a user to select a collection by clicking on the whole row
+    function toggleCollection(selectedCollection) {
+        // If we are dealing with a single post the user can add and remove from collection
+        // For mass updates of many post the user is only permitted to add to collection
+        if ($scope.posts.length === 1) {
+            if (_.contains($scope.posts[0].sets, String(selectedCollection.id))) {
+                $scope.removeFromCollection(selectedCollection);
+            } else {
+                $scope.addToCollection(selectedCollection);
+            }
+        } else {
+            $scope.addToCollection(selectedCollection);
+        }
+    }
+
+    // This function passes flow control to the Collection Creation modal
+    // In Toggle mode: we expect the creation editor to return control to the Listing directive
+    // In the toggleMode we attach the post(s) to the event, collectionEditor will return this post(s)
+    // after the creation, along with the newly created collection
+    function createNewCollection() {
+        $scope.collectionListingVisible = false;
+        $scope.isToggleMode ? $rootScope.$emit('collectionCreate:show', $scope.posts) : $rootScope.$emit('collectionEditor:show');
+    }
+
+    // Search and filter collection list by query term
+    function searchCollections(collectionQuery) {
+        $scope.collections = CollectionEndpoint.editableByMe(
+            {
+                q: collectionQuery
+            }
+        );
+    }
+
+    // Update collection listing when collection are updated elsewhere
+    // in the app
+    $rootScope.$on('collection:update', function () {
+        $scope.loadCollections();
+    });
+
+    // Show listing modal
+    $rootScope.$on('collectionListing:show', function () {
+        $scope.collectionListingVisible = true;
+        $scope.loadCollections();
+    });
+
+    // Show listing modal in Collection Toggle mode
+    // This allows users to select collection to add a given post(s) to
+    // This is called with a post array of 1 or more entries
+    $rootScope.$on('collectionToggle:show', function (event, posts) {
```

Comment:
For future : I've generally be putting event bindings before all other functions. Often as named functions so you can scan the file quickly and find
```
$rootScope.$on('collectionToggle:show', openCollectionToggle);
```